wallet-new/src/Cardano/Wallet/API/Response.hs

```diff
@@ -64,6 +75,22 @@ instance Arbitrary a => Arbitrary (WalletResponse a) where
 instance ToJSON a => MimeRender OctetStream (WalletResponse a) where
     mimeRender _ = encode
 
+instance (ToSchema a, Typeable a) => ToSchema (WalletResponse a) where
+    declareNamedSchema _ = do
+        let a = Proxy @a
+            tyName = toText . show $ typeRep a
+        aRef <- declareSchemaRef a
+        respRef <- declareSchemaRef (Proxy @ResponseStatus)
+        metaRef <- declareSchemaRef (Proxy @Metadata)
+        pure $ NamedSchema (Just $ "WalletResponse<" <> tyName <> ">") $ mempty
+            & type_ .~ SwaggerObject
+            & required .~ ["data", "status", "meta"]
+            & properties .~
+                [ ("data", aRef)
+                , ("status", respRef)
+                , ("meta", metaRef)
+                ]
```

Comment:
This is a trick I've learned at a previous place. Response envelopes get *weird* if you don't name the things differently, since they get repeated, and Swagger doesn't support parameterized templates.